sqlglot/expressions.py

```diff
@@ -637,7 +637,7 @@ def transform(self, fun: t.Callable, *args: t.Any, copy: bool = True, **kwargs)
 
             if not root:
                 root = new_node
-            elif new_node is not node:
+            elif parent and arg_key and new_node is not node:
```

Comment:
Note: this & the CTE changes in parser.py were triggered by a mypy upgrade that [broke CI](https://github.com/tobymao/sqlglot/actions/runs/13154289477/job/36707801411).